### PR TITLE
Update obsolete rubies and add a doc for compatibility

### PR DESCRIPTION
**Pull request by koic**
Follow up https://github.com/rubocop-hq/rubocop/pull/7943#discussion_r422450919 and https://github.com/rubocop-hq/rubocop/issues/8112#issuecomment-640539334.

This PR updates obsolete rubies, adds a doc for compatibility and  replaces compatibility description with compatibility doc link to keep README.md simple.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: docs/modules/ROOT/pages/compatibility.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+= Compatibility
```
Comment (bbatsov) [inline docs/modules/ROOT/pages/compatibility.html:1]: I think you meant to use `.adoc` as the extension here.

**Comment by bbatsov** [comment]
New pages should also be listed in the `nav.adoc`.

**Comment by koic** [comment]
Oops, I forgot it. I updated nav.adoc and tweak README.md.

**Comment by bbatsov** [comment]
I think we should clarify what we mean by compatibility here, as there's the "runtime" aspect and the "analysis" aspect (supported Ruby target version). I'm guessing some people might be confused by the two.

**Comment by koic** [comment]
Sure! That is a practical note. I updated the doc.

File: docs/modules/ROOT/pages/compatibility.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+= Compatibility
+
+RuboCop targets Ruby 2.4+.
+
+RuboCop officially supports CRuby and JRuby. Support for JRuby is based on CRuby.
+
+== Support Matrix
+
+RuboCop has customarily provided support for about one year after EOL of CRuby version. The following table is the support matrix.
```
[the remark this answers is not in the document]
Comment (bbatsov) [reply docs/modules/ROOT/pages/compatibility.adoc:9]: We've been doing it for years, anyways. In most cases this doesn't really require any additional efforts on our part, as lately there haven't been that many changes in Ruby.

File: docs/modules/ROOT/pages/compatibility.adoc [hunk 3]
```diff
@@ -0,0 +1,31 @@
+= Compatibility
+
+RuboCop targets Ruby 2.4+.
+
+RuboCop officially supports CRuby and JRuby. Support for JRuby is based on CRuby.
```
[the remark this answers is not in the document]
Comment (koic) [reply docs/modules/ROOT/pages/compatibility.adoc]: Thanks for the feedback. I updated it. If you're still not sure, I'll try updating further :-)

**Comment by bbatsov** [comment]
Thanks!

**Comment by bbatsov** [comment]
The new page is up https://docs.rubocop.org/rubocop/compatibility.html

**Comment by koic** [comment]
Thank you for the tweak doc and deploying the site docs 😍

**Pull request merged**